docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -0,0 +1,70 @@
+
+## .NET 8 native AOT diagnostic support
+
+The following table summarizes diagnostic features supported for native AOT deployments:
+
+| Feature | Fully supported | Partially supported | Not supported |
+| - | - | - | - |
+| Build (Inner dev loop) diagnostics | <span aria-hidden="true">✔️</span><span class="visually-hidden">Fully supported</span> | | |
+| Observability | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| CPU Profiling | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Production debugging | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| Heap analysis | | | <span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> |
+
+### Build (inner dev loop) diagnostics
+
+Publishing your app as native AOT produces an app that has been ahead-of-time (AOT) compiled to native code. As mentioned above, this means that not all diagnostic tools will work seamlessly with published native AOT applications in .NET 8. However, all .NET diagnostic tools are available for developers during the application building stage. We recommend developing, debugging, and testing the applications as usual and publish the working app with native AOT as one of the last steps.
+
+### Observability
+
+The native AOT runtime supports [EventPipe](../../diagnostics/eventpipe.md), which allows native AOT apps to easily trace their applications. EventPipe support also allows most .NET diagnostic tools, like [dotnet-trace](../../diagnostics/dotnet-trace.md), [dotnet-counter](../../diagnostics/dotnet-counters.md), and [dotnet-monitor](../../diagnostics/dotnet-monitor.md), to work seamlessly with native AOT applications. EventPipe is an optional component in native AOT. To include EventPipe support, set the `EventSourceSupport` MSBuild property to `true`.
+
+```xml
+<PropertyGroup>
+    <EventSourceSupport>true</EventSourceSupport>
+</PropertyGroup>
+```
+
+[OpenTelemetry](../../diagnostics/observability-with-otel.md) is expected to support native AOT in the key three pillars of observability&mdash;logging, tracing, and metrics&mdash;in .NET 8. Native AOT provides partial support for some [well-known event providers](../../diagnostics/well-known-event-providers.md). Not all [runtime events](../../../fundamentals/diagnostics/runtime-events.md) are supported in native AOT.
+
+### CPU profiling
+
+The "Microsoft-DotNETCore-SampleProfiler" provider is not currently supported in native AOT. However, platform-specific tools like [PerfView](https://github.com/microsoft/perfview) and [Perf](https://perf.wiki.kernel.org/index.php/Main_Page) can be used to collect CPU samples of a native AOT application.
```

Comment:
```suggestion
Platform-specific tools like [PerfView](https://github.com/microsoft/perfview) and [Perf](https://perf.wiki.kernel.org/index.php/Main_Page) can be used to collect CPU samples of a native AOT application.
```

> The "Microsoft-DotNETCore-SampleProfiler" provider is not currently supported in native AOT.

It is not clear how this translates into user experience with tools. I do not think we need to mention this specific provider. (We are not listing the precise list of events and providers that work with native AOT.)